tools/scaffolding/templates/gateway/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM node:14-buster as nodebuild
```

Comment:
Does `node:14-buster` already have `yarn` installed? I see in our main `dockerfile` we run our install tool for it. We could be doing extra work in our main `dockerfile` if this is the case.
https://github.com/lyft/clutch/blob/fb1e24c8bf1cedc6dcc1ce066141f99d4ead3c29/Dockerfile#L4

Reply:
Our yarn policy uses yarn from a specific path to avoid version floats. This is required.

Reply:
This gets a bit fiddly. At this stage of the build it is running in a `yarn` container that only includes `frontend/` and we can't copy `tools/install-yarn.sh` from `./` unless we include it in the gateway template.

Thoughts? Should we add `tools/install-yarn.sh` to `templates/gateway/`? Would you rather I just extract the logic from `install-yarn.sh` and run it in the `Dockerfile` directly to avoid having to include `install-yarn.sh` in the template?

Reply:
actually at the moment we don't lock the yarn version in the scaffolded gateway (i.e. no .yarnrc in frontend/). i think for now let's avoid making the change, it's a separate enhancement from just getting the build up and running in the Dockerfile. in reality, we haven't run into any issues with yarn floating yet.

normally in the gateway we access parent scripts from a go module which complicates things in this dockerfile where it's not immediately in context.

easiest way to solve it would likely be to thread a variable through with the proper version and in the dockerfile just make sure it matches.

filed https://github.com/lyft/clutch/issues/472 to track